core/state/statedb.go

```diff
@@ -932,59 +932,65 @@ func (s *StateDB) clearJournalAndRefund() {
 // of a specific account. It leverages the associated state snapshot for fast
 // storage iteration and constructs trie node deletion markers by creating
 // stack trie with iterated slots.
-func (s *StateDB) fastDeleteStorage(snaps *snapshot.Tree, addrHash common.Hash, root common.Hash) (map[common.Hash][]byte, *trienode.NodeSet, error) {
+func (s *StateDB) fastDeleteStorage(snaps *snapshot.Tree, addrHash common.Hash, root common.Hash) (map[common.Hash][]byte, map[common.Hash][]byte, *trienode.NodeSet, error) {
 	iter, err := snaps.StorageIterator(s.originalRoot, addrHash, common.Hash{})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	defer iter.Release()
 
 	var (
-		nodes = trienode.NewNodeSet(addrHash)
-		slots = make(map[common.Hash][]byte)
+		nodes   = trienode.NewNodeSet(addrHash)
+		slots   = make(map[common.Hash][]byte)
+		deletes = make(map[common.Hash][]byte)
 	)
 	stack := trie.NewStackTrie(func(path []byte, hash common.Hash, blob []byte) {
 		nodes.AddNode(path, trienode.NewDeleted())
 	})
 	for iter.Next() {
 		slot := common.CopyBytes(iter.Slot())
 		if err := iter.Error(); err != nil { // error might occur after Slot function
-			return nil, nil, err
+			return nil, nil, nil, err
 		}
-		slots[iter.Hash()] = slot
+		key := iter.Hash()
+		slots[key] = slot
+		deletes[key] = nil
```

Comment:
I don't quite get this. You have added a return-value, `deletes`, which is afaict identical to the `slots`, except that you only ever store `nil` in it. So it's basically the `maps.Keys` of the `slots`. Why?

Reply:
Sorry the variable names are very confusing, I will fix it.

The slot denotes the set which includes all the mutated storage slots along with their **original** value;
The deletes denotes the set which includes all the mutated storage slots along with their **current** value, it's nil here;

Reply:
> So it's basically the maps.Keys of the slots. Why?

Yes, the new set `deletes` denotes the storage changes of the account, which will be aggregated into the global state set later.